File: easy/perl/onboarding.pl
while (1) {
    my $enemy1 = <STDIN>;
    chomp($enemy1);
    my $dist1 = <STDIN>;
    chomp($dist1);
    my $enemy2 = <STDIN>;
    chomp($enemy2);
    my $dist2 = <STDIN>;
    chomp($dist2);

    if ($dist1 < $dist2) {
        print "$enemy1\n";
    } else {
        print "$enemy2\n";
    }
}

File: easy/scala/onboarding.scala
object Main extends App {
  while (true) {
    val enemy1 = scala.io.StdIn.readLine()
    val dist1 = scala.io.StdIn.readInt()
    val enemy2 = scala.io.StdIn.readLine()
    val dist2 = scala.io.StdIn.readInt()

    if (dist1 < dist2) {
      println(enemy1)
    } else {
      println(enemy2)
    }
  }
}
